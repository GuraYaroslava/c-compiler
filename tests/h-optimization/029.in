int main()
{

    int Add(int a, int b)
    {
        return a + b;
    }

    int i = 0;
    while (i == 0)
    {
        printf("one _ start\n");
        int a;
        while (i == 0)
        {
            printf("two _ start\n");
            int b;

            while (0)
            {
                Add(1, 1);
                printf("one\n");
            }

            while (0)
            {
                Add(2, 2);
                printf("two\n");
            }

            while (0)
            {
                Add(3, 3);
                printf("three\n");
            }

            while (i == 0)
            {
                int e;
                printf("while #three\n");
                i -= 1;
            }
            printf("two _ end\n");
        }
        int f;
        printf("one _ end");
    }

    return 0;
}
