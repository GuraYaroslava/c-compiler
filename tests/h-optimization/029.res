one _ start
two _ start
while #three
two _ end
one _ end