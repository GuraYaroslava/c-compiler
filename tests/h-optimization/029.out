table 0:
`int`
`float`
`char`

table 1:
variable `main` type of function() returning `int`
{
variable `i` type of `int`
        i
    = <
        0

while
        i
    == <
        0
{
    function:     printf
        format:     "one _ start\n"


while
        i
    == <
        0
{
    function:     printf
        format:     "two _ start\n"


while
        i
    == <
        0
{
    function:     printf
        format:     "while #three\n"


        i
    -= <
        1

}
    function:     printf
        format:     "two _ end\n"


}
    function:     printf
        format:     "one _ end"


}
return
    0
}


